File: SubSetSum.cpp
#include<iostream>
using namespace std;
void printSolution(int *ar,int n)
{
    cout<<"\n";
    for(int i=0;i<n;i++)
    {
        if(ar[i]!=0)
        {
            cout<<ar[i]<<" ";
        }
    }
}
void solveSumUtil(int *ar,int n,int x,int gs,int calSum,int *ele,int e_size)
{
    if(gs==calSum)
    {
        printSolution(ele,e_size);
        //solveSumUtil(ar,n,x+1,gs,calSum-ar[x],ele,e_size-1);
    }
    else
    {
        for(int i=x;i<n;i++)
        {
            ele[e_size]=ar[i];
            solveSumUtil(ar,n,i+1,gs,calSum+ar[i],ele,e_size+1);
        }
    }
}
void solveSum(int *ar,int n,int gs)
{
    int *ele=new int[n];
    solveSumUtil(ar,n,0,gs,0,ele,0);
}
int main()
{
    cout<<"Enter the value of n:"<<endl;
    int n;
    cin>>n;
    cout<<"Enter the value of array:\n";
    int *ar=new int[n];
    for(int i=0;i<n;i++)
    {
        cin>>ar[i];
    }
    int sum;
    cout<<"Enter the sum :\n";
    cin>>sum;
    solveSum(ar,n,sum);
}

File: SubSetSum.java
import java.io.*;
import java.lang.*;
import java.util.*;
class SubSetSum
{
    public static void printSolution(int ar[],int n)
    {
        for(int i=0;i<n;i++)
        {
            System.out.print(ar[i]+" ");
        }
        System.out.println();
    }
    public static void SSSUtil(int ar[],int n,int gs,int temp[],int x,int sum,int temp_size)
    {
        if(gs==sum)
        {
            printSolution(temp,temp_size);
        }
        else
        {
            for(int i=x;i<n;i++)
            {
                temp[temp_size]=ar[i];
                SSSUtil(ar,n,gs,temp,i+1,sum+ar[i],temp_size+1);
            }
        }
    }
    public static void SSS(int ar[],int n,int gs)
    {
        int temp[]=new int[n];
        int temp_size=0;
        SSSUtil(ar,n,gs,temp,0,0,temp_size);
    }
    public static void main(String args[])throws IOException
    {
        Scanner sc=new Scanner(System.in);
        System.out.println("Enter the number of elements:");
        int n=sc.nextInt();
        int ar[]=new int[n];
        System.out.println("Enter the elements of array:");
        for(int i=0;i<n;i++)
        {
            ar[i]=sc.nextInt();
        }
        System.out.println("Enter the Sum :");
        int gs=sc.nextInt();
        SSS(ar,n,gs);
    }
}